### PR TITLE
atom: fix issues raised by lierdakil

### PR DESCRIPTION
**Pull request by GlenCFL**
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: 
Each change is discussed within the appropriate issue.
#21963
#21964
#21965
#21967
#21969
- [X] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

This pull request addresses all of the issues raised by @lierdakil today.

### CODE REVIEW

**Review by smhxx — APPROVED** [review]
Alright, I'm on board with this. The `Point`-related changes look solid, and I agree that the `Emissions` interface didn't really serve much purpose since it was functionally identical to `Record<any>`. The new implementation is definitely cleaner, imo.

File: types/event-kit/index.d.ts [hunk 1]
```diff
@@ -83,31 +83,40 @@ export class Emitter<Emissions = { [key: string]: any }> implements DisposableLi
...
+    preempt<T extends keyof Emissions>(eventName: T, handler: (value: Emissions[T]) => void):
         Disposable;
 
     // Event Emission
     /** Invoke the handlers registered via ::on for the given event name. */
-    emit<T extends keyof Emissions>(eventName: T, value?: Emissions[T]): void;
+    emit<T extends keyof Emissions>(eventName: T, value: Emissions[T]): void;
+
+    /** Invoke the handlers registered via ::on for the given event name. */
+    emit(eventName: string): void;
```
Comment (lierdakil) [inline types/event-kit/index.d.ts]: Shouldn't these be also keyed by `keyof Emissions` though?

Comment (GlenCFL) [reply types/event-kit/index.d.ts]: Yes. Now that I think about it, we can just keep the old signature here and get rid of the overload. This accomplishes exactly the same thing.

File: types/atom/index.d.ts [hunk 2]
```diff
@@ -2403,27 +2402,25 @@ export interface TextEditorRegistry {
...
-        title?: string,
-        html?: boolean,
-        item?: HTMLElement|{ element: HTMLElement },
-        class?: string,
-        placement?: "top"|"bottom"|"left"|"right"|"auto"|(() => string),
-        trigger?: "click"|"hover"|"focus"|"manual",
-        delay?: { show: number, hide: number },
-        keyBindingCommand?: string,
-        keyBindingTarget?: HTMLElement
+        item?: HTMLElement|{ element: HTMLElement }
```
Comment (lierdakil) [inline types/atom/index.d.ts]: Hmm. I just noticed this was left unchanged. I have to wonder if it's intended or an oversight.

I've mentioned this is seems too strict in passing, but maybe didn't emphasize it enough?

> Also while on topic, I think item can be anything that Atom's view algorithm will get a view for. At least it's usage suggests that: https://github.com/atom/atom/blob/v1.22.1/src/tooltip.js#L342

So... it probably should just be `object`?

Comment (GlenCFL) [reply types/atom/index.d.ts]: It should be. I had read it, but probably just associated it with the other `item` change in the decoration options.

**Pull request merged**